cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/SanityCheck.hs

```diff
@@ -88,3 +95,35 @@ foldBlocksAccumulator _ _ allEvents _ _ =
   filterPoolReap :: LedgerEvent -> Bool
   filterPoolReap (PoolReap _) = True
   filterPoolReap _ = False
+
+
+hprop_asyncRegister_sanity_check :: Property
+hprop_asyncRegister_sanity_check = 
+    withTests 1 . property $ lift $ do 
+
+     beforeForkedThread <- getCurrentTime
+     (internalState,tId) <- runResourceT $ do 
+        s <- getInternalState
+        (_,asyncA) <- asyncRegister_ (threadDelay 10_000_000)
+        let tId = asyncThreadId asyncA
+        return (s,tId)
+     afterForkedThread <- getCurrentTime
```

Comment:
FYI You can use https://hackage-content.haskell.org/package/hedgehog-extras-0.10.1.0/docs/Hedgehog-Extras-Test-TestWatchdog.html to verify if code paths were reached.